src/wristpy/core/orchestrator.py

```diff
@@ -0,0 +1,259 @@
+
+        for field_name, field_value in self:
+            if field_value is not None:
+                if (
+                    use_epoch1_time
+                    and field_name
+                    not in ["enmo", "anglez", "sleep_windows", "nonwear_array"]
+                ) or (not use_epoch1_time and field_name in ["enmo", "anglez"]):
+                    results_dataframe = results_dataframe.with_columns(
+                        pl.Series(field_name, field_value.measurements)
+                    )
+
+        return results_dataframe
+
+
+def format_sleep_data(
+    sleep_windows: List[analytics.SleepWindow], reference_measure: models.Measurement
+) -> np.ndarray:
+    """Formats sleep windows into an array for saving.
+
+    Args:
+        sleep_windows: The list of time stamp pairs indicating periods of sleep.
+        reference_measure: The measure from which the temporal resolution will be taken.
+            Any epoch1 measure can be used, but enmo is used for consistency.
+
+    Returns:
+        1-D binary np.ndarray, with 1 indicating sleep. Will be of the same length as
+            the timestamps in the epoch1_measure.
+    """
+    sleep_array = np.zeros(len(reference_measure.time))
+
+    for window in sleep_windows:
+        sleep_mask = (reference_measure.time >= window.onset) & (
+            reference_measure.time <= window.wakeup
+        )
+        sleep_array[sleep_mask] = 1
+
+    return sleep_array
+
+
+def format_nonwear_data(
+    nonwear_data: models.Measurement, reference_measure: models.Measurement
+) -> np.ndarray:
+    """Format nonwear data into an array for saving.
+
+    Args:
+        nonwear_data: The nonwear measurement.
+        reference_measure: The measure from which the temporal resolution will be taken.
+            Any epoch1 measure can be used, but enmo is used for consistency.
+
+    Returns:
+        1-D binary np.ndarray, with 1 indicating nonwear. Will be of the same length as
+            the timestamps in the reference_measure.
+
+    """
+    nonwear_df = pl.DataFrame(
+        {
+            "nonwear": nonwear_data.measurements.astype(np.int64),
+            "time": nonwear_data.time,
+        }
+    ).set_sorted("time")
+
+    nonwear_upsample = nonwear_df.upsample(time_column="time", every="5s").fill_null(
+        strategy="forward"
+    )
+
+    end_sequence = pl.repeat(
+        nonwear_upsample["nonwear"][-1],
+        n=(len(reference_measure.time) - len(nonwear_upsample["time"])),
+        dtype=pl.Int64,
+        eager=True,
+    )
+    full_nonwear_array = pl.concat([nonwear_upsample["nonwear"], end_sequence])
+
+    return full_nonwear_array.to_numpy()
+
+
+def run(
+    input: pathlib.Path,
+    output: Optional[pathlib.Path] = None,
+    settings: config.Settings = config.Settings(),
+    calibrator: calibration.Calibration = calibration.Calibration(),
+    detect_nonwear_kwargs: Optional[Dict[str, Any]] = None,
```

Comment:
See typing.Unpack https://docs.python.org/3/library/typing.html#typing.Unpack for how to type this.

Reply:
Oh wait, this isnt a real kwargs (no double asterisk), thats some confusing variable naming (: . I'd just replace entirely with a TypedDict and rename it to something like nonwear_settings. 

Alternatively, should these settings be ported to the overall settings @Asanto32 ?

Reply:
Yeah, those all have default values in `detect_nonwear` and I would agree that they could be modified in the `config.Settings`. Especially as it looks like the `std_criteria` and `range_criteria` will need to be set by the readers (based on GENEActiv vs Actigraph). I don't expect the `short_epoch_length` and `n_short_epoch_in_long_epoch` to be touched.

Reply:
I like making these settings, I'll make that change, and make it so the detect_nonwear function gets its default values from Settings.